### PR TITLE
6-create-openapi-doc-file

### PR DESCRIPTION
**Pull request by callumadair**

### CODE REVIEW

File: docs/api.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      summary: Root endpoint for financial management.
+    post:
+      summary: Endpoint to create a new financial planning category.
+  /finance/{id}:
+    get:
+      summary: Endpoint for viewing the state of financial planning category with given id.
+    delete:
+      summary: Endpoint to delete an existing financial planning category.
+    patch:
+      summary: Endpoint to update an existing financial planning category.
```
Comment (Toby-Rea) [inline docs/api.yaml:33]: Should nest these under a `categories` path i.e. `/finance/categories/{id}`

Comment (callumadair) [reply docs/api.yaml:33]: thought we only had categories here though?

Comment (callumadair) [inline docs/api.yaml:33]:
```suggestion
  /finance/category/{id}:
    get:
      summary: Endpoint for viewing the state of financial planning category with given id.
    delete:
      summary: Endpoint to delete an existing financial planning category.
    patch:
      summary: Endpoint to update an existing financial planning category.
```

Comment (Toby-Rea) [reply docs/api.yaml:33]: I don't want us to get walled in if we choose to extend this e.g. adding things like savings pots, etc...

**Review by Toby-Rea — APPROVED** [review]

**Pull request merged**